### PR TITLE
Fixing pow assumptions with regard to rationals

### PR DESCRIPTION
**Pull request by pelegm**
This fixes bullets 4 and 5, re #8046.

### CODE REVIEW

File: sympy/core/power.py [hunk 1]
```diff
@@ -969,7 +969,11 @@ def _eval_is_rational(self):
             return False
         if e.is_integer:
             if b.is_rational:
-                return True
+                if b.is_nonzero or e.is_nonnegative:
+                    return True
+                # In the special case of b**b, the result is rational
```
Comment (skirpichev) [inline sympy/core/power.py]: Why this comment?

File: sympy/core/tests/test_assumptions.py [hunk 2]
```diff
@@ -767,7 +768,8 @@ def test_special_is_rational():
     assert (sqrt(3) + S.Pi).is_rational is False
     assert (x**i).is_rational is None
     assert (i**i).is_rational is True
-    assert (r**i).is_rational is True
+    assert (i**i2).is_rational is None
+    assert (r**i).is_rational is None
```
Comment (skirpichev) [inline sympy/core/tests/test_assumptions.py]: Please add tests from the issue.

File: sympy/core/power.py (hunk 1 above)
Comment (pelegm) [inline sympy/core/power.py]: It's explaining the code. What's wrong with that?

File: sympy/core/tests/test_assumptions.py (hunk 2 above)
Comment (pelegm) [inline sympy/core/tests/test_assumptions.py]: Sure, I'll do that.

File: sympy/core/power.py (hunk 1 above)
Comment (skirpichev) [inline sympy/core/power.py]: Why do you think it does explain anything?  This is just verbose equivalent of the following lines.

Perhaps, you want to explain why case `b**b` is rational.

Comment (pelegm) [inline sympy/core/power.py]: Forget it, I'll just remove it.

Comment (smichr) [inline sympy/core/power.py]: I like a comment, but maybe `if b == e:  # always rational, even for 0**0`

File: sympy/core/tests/test_assumptions.py (hunk 2 above)
Comment (smichr) [inline sympy/core/tests/test_assumptions.py]: The generic tests that are here cover the code added, don't they? If so, this is fine.

File: sympy/core/power.py (hunk 1 above)
Comment (smichr) [inline sympy/core/power.py]: Could you make the change I recommend? Then this is ready to go.

**Comment by pelegm** [comment]
@smichr, fixed the comment, I agree the generic tests cover the added code. Regarding tests from the issue, perhaps we should add them, I am not sure. I'll let you decide. We can also add these when closing the issue, whenever that will occur.

**Comment by smichr** [comment]
Thanks, it's in.  I'll check off the bullets on the issue, too.

**Pull request merged**